src/node.cpp

```diff
@@ -377,6 +378,13 @@ int main(int argc, char * argv[]) {
                              angle_min, angle_max, max_distance,
                              frame_id);
             }
+        } else {
+            rplidar_response_device_health_t healthinfo;
```

Comment:
Just curious, can you add a print statement here and check if this else statement ever gets called when the Lidar is properly connected? Also I would suggest to add print statement `ROS_WARN_THROTTLE(1, "RPLIDAR disconnected. Attempting reconnection.")` to help us log this information.

Reply:
@gauthamm addressed comments, node no longer terminates if not connected